File: src/main/java/org/xi/quick/codegenerator/mapper/ColumnsMapper.java
package org.xi.quick.codegenerator.mapper;

import org.apache.ibatis.annotations.Mapper;
import org.apache.ibatis.annotations.Param;
import org.xi.quick.codegenerator.entity.Column;

import java.util.List;

@Mapper
public interface ColumnsMapper {

    /**
     * 获取表的列
     *
     * @param databaseName
     * @param tableName
     * @return
     */
    List<Column> getColumns(@Param("databaseName") String databaseName, @Param("tableName") String tableName);
}

File: code-generator-kt/src/main/kotlin/org/xi/quick/codegeneratorkt/mapper/ColumnsMapper.kt
package org.xi.quick.codegeneratorkt.mapper

import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Param
import org.xi.quick.codegeneratorkt.entity.Column

@Mapper
interface ColumnsMapper {

    /**
     * 获取表的列
     *
     * @param databaseName
     * @param tableName
     * @return
     */
    fun getColumns(@Param("databaseName") databaseName: String, @Param("tableName") tableName: String): List<Column>

    /**
     * 获取表的列
     *
     * @param databaseName
     * @param tableName
     * @return
     */
    fun getColumnsWithIndex(@Param("databaseName") databaseName: String, @Param("tableName") tableName: String): List<Column>
}
